isisd/isis_cli.c

```diff
@@ -279,25 +279,26 @@ DEFPY(no_ip_router_isis, no_ip_router_isis_cmd,
       "IS-IS routing protocol\n"
       "Routing process tag\n")
 {
-	const struct lyd_node *dnode =
-		yang_dnode_get(running_config->dnode, VTY_CURR_XPATH);
+	const struct lyd_node *dnode;
 
-	/* if both ipv4 and ipv6 are off delete the interface isis container too
+	dnode = yang_dnode_get(vty->candidate_config->dnode,
+			       "%s/frr-isisd:isis", VTY_CURR_XPATH);
+	if (!dnode)
+		return CMD_SUCCESS;
+
+	/*
+	 * If both ipv4 and ipv6 are off delete the interface isis container.
 	 */
-	if (!strncmp(ip, "ipv6", strlen("ipv6"))) {
-		if (dnode
-		    && !yang_dnode_get_bool(dnode,
-					    "./frr-isisd:isis/ipv4-routing"))
+	if (strmatch(ip, "ipv6")) {
```

Comment:
maybe a nit here but most people (ie me) will read the strmatch and ipv6 and expect the true part of the statement to be v6 handling not v4 handling.  Worth thinking about?

Reply:
You didn't read it wrong, the true part of the statement is v6 handling. What might have confused you is that the code is checking if the `"./ipv4-routing"` leaf is false too so that the entire `frr-isisd:isis` parent container can be removed, otherwise `"./frr-isisd:isis/ipv6-routing"` is set to false.